### PR TITLE
Design doc of fixed-point quantization.

### LINKED ISSUE #10552: Write fixed-point quantization design.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/10552

You also can see https://github.com/qingqing01/Paddle/blob/quantization_doc/doc/fluid/design/quantization/fixed_point_quantization.md

### CODE REVIEW

File: doc/fluid/design/quantization/fixed_point_quantization.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+       &=(\frac{X_q}{n - 1} * X_m) * (\frac{W_q}{n - 1} * W_m) 
+\end{align}
+$$
+
+From these formulas, dequantization also can be moved before GEMM, do dequantization for $Xq$ and $Wq$ at first, then do GEMM. The forward workflow in training is equivalent to following framework.
+
+<p align="center"> 
+<img src="quantization_forward.png"  width="300" height="330"  /><br/>
+
+Fig 2. Equitvalent forward in training with simulated quantization.
```
Comment (kuke) [inline doc/fluid/design/quantization/fixed_point_quantization.md]: ```Equitvalent``` --> ```Equivalent```

File: doc/fluid/design/quantization/fixed_point_quantization.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+
+How to calculate the quantization range (or maximum absolute value) for inference will be described in the last part.
+
+
+### Training Framework
+
+#### Forward pass
+
+The forward pass is simulated quantization, see the figure 1.
```
Comment (kuke) [inline doc/fluid/design/quantization/fixed_point_quantization.md]: ```see the figure 1``` --> ```see Figure 1```

File: doc/fluid/design/quantization/fixed_point_quantization.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+Fig 2. Equitvalent forward in training with simulated quantization.
+
+</p>
+
+We use this equivalent workflow in the training. In our desigin, there is a quantization transipler to insert the quantization operator and the de-quantization operator in the Fluid `ProgramDesc`.
+
+#### Backward pass
+
+See the figure 3. The gradients are calculated by dequantized weights and activations. All inputs and outputs are float point with 32 bit. And in the weight updating process, the gradients will be added to the original weight, not the quantized or dequantized weights.
```
Comment (kuke) [inline doc/fluid/design/quantization/fixed_point_quantization.md]: ```See the figure 3``` --> ```See Figure 3```

File: doc/fluid/design/quantization/fixed_point_quantization.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+<p align="center"> 
+<img src="quantization_backward_and_optimization.png" /><br/>
+
+Fig 3. Backward and weight updating in training with simulated quantization.
+
+</p>
+
+So the quantization transipler will change some inputs of the corresponding backward operators. 
+
+### How to calculate quantization scale
```
Comment (kuke) [inline doc/fluid/design/quantization/fixed_point_quantization.md]: ```scale``` or ```range```, should they be consistent?

File: doc/fluid/design/quantization/fixed_point_quantization.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+There are two strategies to calculate quantization scale, we call them dynamic and static strategy. The dynamic strategy is to calculate the quantization scale value each iteration. The static strategy is to fix the quantization scale for different inputs.
```
Comment (kuke) [inline doc/fluid/design/quantization/fixed_point_quantization.md]: ```is to calculate``` --> ```calculates```
```is to fix the quantization scale``` --> ```keeps the same quantization scale```

File: doc/fluid/design/quantization/fixed_point_quantization.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+For weights, we apply the dynamic strategy for weights in the training, that is to say, the quantization scale will recalculate during each iteration until the traning is finished.
```
Comment (kuke) [inline doc/fluid/design/quantization/fixed_point_quantization.md]: remove ```for weights```
```will recalculate``` --> ```will be recalculated```

File: doc/fluid/design/quantization/fixed_point_quantization.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+For activations, the quantization scales are estimated during training, then use them in inference. There are several different ways to estimat:
```
Comment (kuke) [inline doc/fluid/design/quantization/fixed_point_quantization.md]: ```then use them``` --> ```then used```
```estimat``` --> ```estimate them```

**Review by kuke — COMMENTED** [review]
Thanks for the design doc. Some comments about the writing

**Review by qingqing01 — COMMENTED** [review]
@kuke @panyx0718  Thanks your detailed review. Thanks very much.

**Comment by qingqing01** [comment]
@panyx0718 @kuke  Is there any problem about this PR ?

**Review by kuke — APPROVED** [review]
LGTM

**Pull request merged**